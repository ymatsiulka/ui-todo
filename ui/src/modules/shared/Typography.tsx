import React from 'react';

type TypographyVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

interface TypographyProps {
  variant: TypographyVariants;
  children?: React.ReactNode;
  className?: string;
  text?: string;
}

const Typography: React.FC<TypographyProps> = ({ variant, children, className, text, ...props }) => {
  const TypographyComponent = variant;
  return (
    <TypographyComponent {...props} className={className}>
      {text ?? children}
    </TypographyComponent>
  );
};

export default Typography;
