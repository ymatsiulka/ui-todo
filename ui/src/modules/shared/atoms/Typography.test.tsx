import React from 'react';
import { render } from '@testing-library/react';
import Typography from './Typography';

test('Typography should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(<Typography variant="h1" text="123" className="spacer"></Typography>);

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
