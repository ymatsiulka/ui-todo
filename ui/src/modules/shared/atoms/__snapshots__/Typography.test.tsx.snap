// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render properly 1`] = `
<DocumentFragment>
  <h1
    class="spacer"
  >
    123
  </h1>
</DocumentFragment>
`;
